presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -1214,31 +1214,13 @@ private PlanWithProperties arbitraryDistributeUnion(
         @Override
         public PlanWithProperties visitApply(ApplyNode node, Context context)
         {
-            PlanWithProperties input = node.getInput().accept(this, context);
-            PlanWithProperties subquery = node.getSubquery().accept(this, context.withCorrelations(node.getCorrelation()));
-
-            ApplyNode rewritten = new ApplyNode(
-                    node.getId(),
-                    input.getNode(),
-                    subquery.getNode(),
-                    node.getSubqueryAssignments(),
-                    node.getCorrelation());
-            return new PlanWithProperties(rewritten, deriveProperties(rewritten, ImmutableList.of(input.getProperties(), subquery.getProperties())));
+            throw new IllegalStateException("Unexpected node: " + node.getClass().getName());
```

Comment:
- no, it is also used as intermediate checker before any optimizer is run. However `NoSubqueryRelatedNodeLeftCheck` becomes obsolete, but this I would like to tackle in regards of https://github.com/prestodb/presto/issues/6580

 - `LateralJoinNode` does not have `toString` so that way the message is cleaner